repo/fsrepo/datastores.go

```diff
@@ -136,6 +147,10 @@ func FlatfsDatastoreConfig(params map[string]interface{}) (DatastoreConfig, erro
 	return &c, nil
 }
 
+func (c *flatfsDatastoreConfig) DiskId() string {
+	return fmt.Sprintf("flatfs;%s;%s", c.path, c.shardFun.String())
```

Comment:
I am not sure if including path is right thing to do.
Usually when path is changed the user will migrate data himself.